src/modules/mc_rate_control/MulticopterRateControl.cpp

```diff
@@ -147,12 +145,18 @@ MulticopterRateControl::Run()
 	vehicle_angular_velocity_s angular_velocity;
 
 	if (_vehicle_angular_velocity_sub.update(&angular_velocity)) {
+
+		// grab corresponding vehicle_angular_acceleration immediately after vehicle_angular_velocity copy
```

Comment:
We may document on both sides - _mc_rate_control and vehicle_angular_velocity - that velocity is published before accel, and that they should be read in the same order. This is not obvious and may be lost in a future refactoring.

Reply:
The ordering doesn't actually matter because the execution of `vehicle_angular_velocity` (where both angular velocity and angular acceleration are published) followed by `mc_rate_control` is serialized. Although there's nothing actually enforcing this implementation.

I'll still add a comment publisher side.

Reply:
Right, 👍